src/vm.rs

```diff
@@ -67,18 +69,26 @@ impl EraVM {
     }
 
     /// Run a vm program with a given bytecode.
-    pub fn run_program_with_custom_bytecode(&mut self) -> ExecutionOutput {
+    pub fn run_program_with_custom_bytecode(&mut self) -> (ExecutionOutput, BlobSaverTracer) {
         self.run_opcodes()
     }
 
-    pub fn run_program_with_test_encode(&mut self) -> ExecutionOutput {
-        self.run(&mut [], EncodingMode::Testing)
-            .unwrap_or(ExecutionOutput::Panic)
+    pub fn run_program_with_test_encode(&mut self) -> (ExecutionOutput, BlobSaverTracer) {
+        let mut tracer = BlobSaverTracer::new();
+        let r = self
+            .run(&mut [Box::new(&mut tracer)], EncodingMode::Testing)
+            .unwrap_or(ExecutionOutput::Panic);
+
+        (r, tracer)
     }
 
-    fn run_opcodes(&mut self) -> ExecutionOutput {
-        self.run(&mut [], EncodingMode::Production)
-            .unwrap_or(ExecutionOutput::Panic)
+    fn run_opcodes(&mut self) -> (ExecutionOutput, BlobSaverTracer) {
+        let mut tracer = BlobSaverTracer::new();
+        let r = self
+            .run(&mut [Box::new(&mut tracer)], EncodingMode::Production)
+            .unwrap_or(ExecutionOutput::Panic);
+
+        (r, tracer)
```

Comment:
Instead of creating the tracer and returning it, the idea is passing any tracer you want from outside and the consuming it if needed, that means,  passing the tracer from compiler tester